widget/embedded/src/utils/quote.ts

```diff
@@ -326,3 +290,98 @@ export const getUniqueBlockchains = (steps: Step[]) => {
 
   return result;
 };
+
+export const sortQuotesBy = (
+  strategy: PreferenceType,
+  quotes: MultiRouteResponse['results']
+): MultiRouteResponse['results'] => {
+  return quotes.sort((quote1, quote2) => {
+    const getScore = (route: MultiRouteSimulationResult) =>
+      route.scores?.find((score) => score.preferenceType === strategy)?.score ??
+      0;
+
+    const quote1Score = getScore(quote1);
+    const quote2Score = getScore(quote2);
+
+    if (quote1Score !== quote2Score) {
+      return quote2Score - quote1Score;
+    }
+
+    const lowerQuoteId1 = quote1.requestId.toLowerCase();
+    const lowerQuoteId2 = quote2.requestId.toLowerCase();
+
+    return lowerQuoteId1.localeCompare(lowerQuoteId2);
+  });
+};
+
+export const getDefaultQuote = (
+  currentQuote: SelectedQuote | null,
+  quotes: MultiRouteSimulationResult[],
+  requestAmount: string
+): SelectedQuote => {
+  if (!currentQuote) {
+    // Handle the case where currentQuote is null
+    return {
+      requestAmount: requestAmount,
+      ...quotes[0], // Return the first quote from the quotes array
```

Comment:
What will happen if quotes array is empty here?

Reply:
I added a condition to state that if the quote was empty, it would return null.